dcrrates/server/tls.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019, The Decred developers
+// See LICENSE for details.
+
+package main
+
+import (
+	"crypto/elliptic"
+	"crypto/tls"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/decred/dcrd/certgen"
+	credentials "google.golang.org/grpc/credentials"
+)
+
+// openRPCKeyPair creates or loads the RPC TLS keypair specified by the
+// application config.
+func openRPCKeyPair(certPath, keyPath string) (credentials.TransportCredentials, error) {
+	// Generate a new keypair when the key is missing.
+	_, e := os.Stat(keyPath)
+	keyExists := !os.IsNotExist(e)
+	if !keyExists {
+		cert, err := generateRPCKeyPair(certPath, keyPath, certWriter{})
```

Comment:
One thing that would be helpful in this function (either way `cert` is created) is to show the hostnames that the cert is valid for.